gdx/src/com/badlogic/gdx/math/spatial/octree/OctreeNode.java

```diff
@@ -0,0 +1,116 @@
+package com.badlogic.gdx.math.spatial.octree;
+
+import com.badlogic.gdx.math.Vector3;
+import com.badlogic.gdx.math.collision.BoundingBox;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.badlogic.gdx.math.spatial.octree.Octree.*;
+
+public class OctreeNode {
+
+    protected int depth = 0;
+
+    // Represent the boundary of the BoundingBox
+    protected BoundingBox box;
+
+    protected OctreeNode[] children;
+    private Set<Object> geometries = null;
+
+    public OctreeNode(Vector3 min, Vector3 max) {
+        this(min.x, min.y, min.z, max.x, max.y, max.z);
+    }
+
+    public OctreeNode(float x1, float y1, float z1, float x2, float y2, float z2) {
```

Comment:
This method can be useful because of children initialization:
```
children[TopLeftFront] = new OctreeNode(
            box.min.x, midy, midz,
            midx, box.max.y, box.max.z)
            .depth(depth + 1);
```